core/src/gl/shaderProgram.cpp

```diff
@@ -84,15 +120,20 @@ GLint ShaderProgram::getAttribLocation(const std::string& _attribName) {
 
 }
 
-GLint ShaderProgram::getUniformLocation(const std::string& _uniformName) {
+GLint ShaderProgram::getUniformLocation(const UniformEntries::UniformEntry* _entry) {
```

Comment:
A better approach would be to also cache the real uniform location on the call site, this would require that UniformLocation<s> are unique for each shader. This should be easy to do except for Lights. For Lights injectOnProgram could return a struct of UniformLocations in the call light->injectOnProgram(*m_shaderProgram) for each shader program